### PR TITLE
Fix some new PVS-Studio issues

### PR DESCRIPTION
**Pull request by dimhotepus**
* Fix some remaining issues found by PVS-Studio.
* Fix some issues found by Clang tooling (clang-tidy / clang flow analyzer / ...).
* Fix build error on *nix due to missed `<stddef.h>` for `ptrdiff_t`. 
* Add continuous integration support via Github actions (has some [limits](https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits), but still better than nothing).

Product page: https://www.viva64.com/en/pvs-studio/
How to use for free on open source projects: https://www.viva64.com/en/b/0600/

### CODE REVIEW

File: gstats/gstats.c [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ static int RecvSessionKey()
 			return GE_DATAERROR; //temp fix in case len == -1, SOCKET_ERROR
 	}
 
-	resp[len - 1] = '\0';
+	resp[len] = '\0';
```
Comment (dimhotepus) [inline gstats/gstats.c:1012]: Looks like i made mistake and _**truncate**_ last byte every time.

Now should work as expected with 128 bytes max.

File: Chat/chatSocket.c [hunk 2]
```diff
@@ -12,6 +12,7 @@
 *************/
 #include <stdio.h>
 #include <stdarg.h>
+#include <stddef.h>
```
Comment (dimhotepus) [inline Chat/chatSocket.c:15]: Buildfix

**Comment by dimhotepus** [comment]
@arves100 Could you please take a look?

**Comment by arves100** [comment]
> @arves100 Could you please take a look?

will review them in some days, I was pretty unactive in this first days of august

**Comment by dimhotepus** [comment]
> > @arves100 Could you please take a look?
> 
> will review them in some days, I was pretty unactive in this first days of august

No problem, we are not in a hurry. Thank you for review.

**Review by arves100 — CHANGES_REQUESTED** [review]

**Comment by arves100** [comment]
I'm terribly sorry for the delay in reviewing this commits, thank you for sending this pr and your patience! there are two issues there:
https://github.com/GameProgressive/UniSpySDK/pull/28/commits/0b8853ce6c8bad6fbdae2e9e1fbb03d1c649c176
https://github.com/GameProgressive/UniSpySDK/pull/28/commits/0f150812bf09331b48b76694d90a671de07d0be9

for the Github Actions I'm not sure how that works at the moment, I would prefer you sending another PR for migration to GitHub action if is possible so I could review it once I figure out how to use it.

File: ghttp/ghttpEncryption.c [hunk 3]
```diff
@@ -177,8 +177,9 @@ typedef struct gsOpenSSLInterface
 int verify_callback(int preverify, X509_STORE_CTX *x509_ctx)
 {
 	/* For error codes, see http://www.openssl.org/docs/apps/verify.html  */
-
+#ifdef GSI_COMMON_DEBUG
```
[the remark this answers is not in the document]
Comment (dimhotepus) [reply ghttp/ghttpEncryption.c:180]: @arves100  `gsDebugFormat` is defined either as empty macro when `GSI_COMMON_DEBUG` is not defined, or as actual debug format if `GSI_COMMON_DEBUG` is defined,  So both Release and Debug builds are ok.  Should i still wrap `gsDebugFormat` ?

**Comment by dimhotepus** [comment]
> I'm terribly sorry for the delay in reviewing this commits, thank you for sending this pr and your patience! there are two issues there:
> [0b8853c](https://github.com/GameProgressive/UniSpySDK/commit/0b8853ce6c8bad6fbdae2e9e1fbb03d1c649c176)
> [0f15081](https://github.com/GameProgressive/UniSpySDK/commit/0f150812bf09331b48b76694d90a671de07d0be9)
> 
> for the Github Actions I'm not sure how that works at the moment, I would prefer you sending another PR for migration to GitHub action if is possible so I could review it once I figure out how to use it.

Removed Github Actions, wrote a note about `gsDebugFormat`.

File: ghttp/ghttpEncryption.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (arves100) [reply ghttp/ghttpEncryption.c:180]: GSI_COMMON_DEBUG is only be defined when there is a Debug build, Release would not include it. gsDebugFormat will be defined as an empty macro when a release build is happening, but guarding gsDebugFormat definitly doesn't hurt (and I would prefer to do so if it uses some variables like the ssl error)

**Comment by dimhotepus** [comment]
> I'm terribly sorry for the delay in reviewing this commits, thank you for sending this pr and your patience! there are two issues there: [0b8853c](https://github.com/GameProgressive/UniSpySDK/commit/0b8853ce6c8bad6fbdae2e9e1fbb03d1c649c176) [0f15081](https://github.com/GameProgressive/UniSpySDK/commit/0f150812bf09331b48b76694d90a671de07d0be9)
> 
> for the Github Actions I'm not sure how that works at the moment, I would prefer you sending another PR for migration to GitHub action if is possible so I could review it once I figure out how to use it.

Fixed PR comments.

**Review by arves100 — APPROVED** [review]

**Comment by arves100** [comment]
Thanks for the pr!

**Pull request merged**